src/websocket/websocket-server.ts

```
@@ -0,0 +1,1027 @@
+  private createWebSocket(
+    socket: Socket,
+    req: IncomingMessage
+  ): NextRushWebSocket {
+    const id = crypto.randomUUID();
+    const ip =
+      (req.headers['x-forwarded-for'] as string)?.split(',')[0] ||
+      req.socket.remoteAddress ||
+      '127.0.0.1';
+
+    // 🚀 Event emitter for individual WebSocket connections
+    const eventHandlers = new Map<string, Function[]>();
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)